README.md

```diff
@@ -61,22 +82,53 @@ def my_task(x, y):
     return x + y
 ```
 
-or use a class object to customize behavior scoped to specific tasks:
+Alternatively, specifying a `SlurmExecutor` instance allows users customize behavior scoped to specific tasks. The `prerun_commands` parameter can be used here to provide a list of shell commands to execute before submitting the workflow. Similarly, the `postrun_commands` parameter takes a list of shell commands to execute *after* submitting the script. Finally, the `srun_append` parameter can be used to modify the call to `srun` that submits the workflow, by adding commands outside the scope of `srun`'s options.
+
+An example of a custom executor instance is shown below:
 
 ```python
 executor = ct.executor.SlurmExecutor(
     remote_workdir="/scratch/user/experiment1",
     options={
-        "partition": "compute",
-	"cpus-per-task": 8
-    }
+        "nodes": 1,
+	    "cpus-per-task": 8,
+        "qos": "regular"
+    },
+    srun_options={
+        "slurmd-debug": 4,
+        "cpu_bind": "cores"
+    },
+    srun_append="nsys profile --stats=true -t cuda --gpu-metrics-device=all",
+    prerun_commands = [
```

Comment:
@santoshkumarradha `ct.DepsBash` commands are issued from the unpickled workflow (say, `script-dispatch_id-0.py`).

(1) If the user needs to execute any *other* `srun` commands (common practice), then these would be nested inside the outer `srun` command that runs the unpickled workflow. I.e. this:
```sh
srun <srun_options> python script-dispatch_id-0.py
```
in the case of a `ct.DepsBash` issuing other `srun` commands, effectively becomes something like this:
```sh
srun <srun_options> srun <options_from_depsbash_command> python unpickled_function.py
```
because `script-dispatch_id-0.py` is where we run any electron dependencies. Nesting `srun` commands is certainly non-standard and problematic. 

(2) `ct.DepsBash` could in principle be used to run the `module load package/1.2.3` command or the `python ./path/to/my/post_process.py -j $SLURM_JOB_ID` command shown here, however. In these cases, the `prerun_commands` and `postrun_commands` feature serve to avoid a second, less common issue with using electron dependencies on Slurm-managed cluster:  If the electron itself is parallelized (with say, `joblib` or `mpi4py`) then `ct.DepsBash` dependencies could be executed multiple times, depending on the resource allocation settings for `srun` (i.e. the `srun_options` parameter).

I think the first issue regarding nested `srun` commands is enough incentive to include this optional feature. I can get back to you with a minimal example of the less common duplicate dependencies issue (2).

Reply:
I think (1) is a good reason to go !.